fews_3di/simulation.py

```diff
@@ -300,36 +302,38 @@ def _add_laterals(self, laterals):
             if not still_to_process:
                 return
 
-    def _add_initial_state(self, saved_state_id_file: Path):
+    def _add_initial_state(self, saved_state_id_file: Path, cold_state_id_file: Path):
         # TODO explain rationale. (likewise for the other methods).
-        if not saved_state_id_file.exists():
-            msg = f"Saved state id file {saved_state_id_file} not found"
-            if self.allow_missing_saved_state:
-                logger.warn(msg)
-                return
-            else:
-                raise utils.MissingFileException(msg)
-        saved_state_id: str = saved_state_id_file.read_text().strip()
-        logger.info("Simulation will use initial state %s", saved_state_id)
-        try:
-            self.simulations_api.simulations_initial_saved_state_create(
-                self.simulation_id, data={"saved_state": saved_state_id}
-            )
-        except openapi_client.exceptions.ApiException as e:
-            if e.status == 400:
-                logger.debug("Saved state setting error: %s", str(e))
-                msg = (
-                    f"Setting initial state to saved state id={saved_state_id} failed. "
-                    f"The error response was {e.body}, perhaps use "
-                    f"--allow-missing-saved-state initially?"
-                )
+        for state_file in [saved_state_id_file,cold_state_id_file]:
+            if not state_file.exists():
+                msg = f"Saved state id file {state_file} not found"
                 if self.allow_missing_saved_state:
```

Comment:
Gaat deze setting erover of er een state file mist? Of dat het toegestaan is dat een staat (niet) meer bestaat op de server?
En zijn ze allebei verplicht?

Reply:
deze setting gaat over het missen van de state file als bestandje op de computer waar de adapter draait.  Later wordt er gekeken via de try except of de state nog beschikbaar is op de 3di server.